### PR TITLE
DEVEXP-352: View results tab below editor tab

### PR DESCRIPTION
**Pull request by BillFarber**
1) This creates a Webview in the VSCode panel (the set of windows at the bottom of the application.

2) The Webview displays HTML.

3) Eval and query results are formatted for HTML and then displayed in the Webview.

### CODE REVIEW

File: client/extension.ts [hunk 1]
```diff
@@ -147,14 +149,23 @@ export function activate(context: vscode.ExtensionContext): void {
     XqyDebugManager.registerMlxprsStatusBarItem(mlxprsStatus);
     JsDebugManager.registerMlxprsStatusBarItem(mlxprsStatus);
     mlxprsStatus.requestUpdate();
+
+
+    const pmbWebViewProvider = new MlxprsWebViewProvider(context.extensionUri);
```
Comment (rjrudin) [inline client/extension.ts]: "pmb" = Bill M. Farber?

File: package.json [hunk 2]
```diff
@@ -378,8 +396,14 @@
                 "type": "object",
                 "title": "MarkLogic Miscellaneous Configuration",
                 "properties": {
-                    "marklogic.sparqlContentType": {
+                    "marklogic.resultsRight": {
                         "order": 1,
+                        "type": "boolean",
+                        "default": false,
+                        "description": "Display query results in an editor tab to the right."
```
Comment (rjrudin) [inline package.json]: Is it guaranteed to be to the right? I'm wondering if it's safer to say "Display query results in a separate editor."

File: client/editorQueryEvaluator.ts [hunk 3]
```diff
@@ -218,4 +281,78 @@ export class EditorQueryEvaluator {
...
+    private static requestMlxprsWebviewUpdateWithRecordArray(response: Record<string, unknown>[], contentType?: string): void {
+        if (this.mlxprsWebViewProvider) {
+            const stringResults =
+                (response as Record<string, unknown>[]).map(record => this.convertRecordToHtml(record, contentType)).join('\n');
+            this.mlxprsWebViewProvider.updateViewContent(stringResults);
+        }
+    }
+
+    private static convertRecordToHtml(record: Record<string, any>, contentType?: string): string {
+        if (record['format'] === 'xml') {
```
Comment (rjrudin) [inline client/editorQueryEvaluator.ts:315]: This all makes sense, but I'm wondering - what is a Record, and what are all the possible keys in it? Like why would "head" be populated but not "value"? Is this all VSCode stuff, or is it stuff we control?

**Review by rjrudin — COMMENTED** [review]
Looks pretty good, and I think it would be fine to open a separate bug for the issue where the MLXPRS window doesn't work until a user clicks on it.

File: client/editorQueryEvaluator.ts (hunk 3 above)
Comment (BillFarber) [reply client/editorQueryEvaluator.ts:315]: I don't have a complete understanding of Record. It is defined in a VSCode library, but it is used to define many of the responses from MarkLogic's Node client. 
These type definitions is an area that could probably benefit from more refactoring and documentation.

**Comment by BillFarber** [comment]
> Looks pretty good, and I think it would be fine to open a separate bug for the issue where the MLXPRS window doesn't work until a user clicks on it.

Bug created: https://project.marklogic.com/jira/browse/DEVEXP-382

**Pull request merged**